src/main/java/com/minecolonies/coremod/placementhandlers/MinecoloniesPlacementHandlers.java

```diff
@@ -0,0 +1,303 @@
+          @NotNull final IBlockState blockState,
+          @Nullable final NBTTagCompound tileEntityData,
+          final boolean complete,
+          final BlockPos centerPos)
+        {
+            final IBlockState newBlockState = BlockUtils.getSubstitutionBlockAtWorld(world, pos);
+            if (complete)
+            {
+                if (!world.setBlockState(pos, blockState, UPDATE_FLAG))
+                {
+                    return ActionProcessingResult.DENY;
+                }
+            }
+            else
+            {
+                if (!world.setBlockState(pos, newBlockState, UPDATE_FLAG))
+                {
+                    return ActionProcessingResult.DENY;
+                }
+            }
+
+            return newBlockState;
+        }
+
+        @Override
+        public List<ItemStack> getRequiredItems(@NotNull final World world, @NotNull final BlockPos pos, @NotNull final IBlockState blockState, @Nullable final NBTTagCompound tileEntityData, final boolean complete)
+        {
+            final IBlockState newBlockState = BlockUtils.getSubstitutionBlockAtWorld(world, pos);
+            final List<ItemStack> itemList = new ArrayList<>();
+            itemList.add(BlockUtils.getItemStackFromBlockState(newBlockState));
+            return itemList;
+        }
+    }
+
+    public static class ChestPlacementHandler implements IPlacementHandler
+    {
+        @Override
+        public boolean canHandle(@NotNull final World world, @NotNull final BlockPos pos, @NotNull final IBlockState blockState)
+        {
+            return blockState.getBlock() instanceof BlockChest;
+        }
+
+        @Override
+        public Object handle(
+          @NotNull final World world,
+          @NotNull final BlockPos pos,
+          @NotNull final IBlockState blockState,
+          @Nullable final NBTTagCompound tileEntityData,
+          final boolean complete,
+          final BlockPos centerPos)
+        {
+            if (tileEntityData != null)
+            {
+                handleTileEntityPlacement(tileEntityData, world, pos);
+            }
+
+            final TileEntity entity = world.getTileEntity(pos);
+            final Colony colony = ColonyManager.getClosestColony(world, pos);
+            if (colony != null && entity instanceof TileEntityChest && colony.getBuildingManager().getBuilding(centerPos) instanceof BuildingWareHouse)
+            {
+                BuildingWareHouse.handleBuildingOverChest(pos, (TileEntityChest) entity, world);
+            }
+
+            if (!world.setBlockState(pos, blockState, UPDATE_FLAG))
+            {
+                return ActionProcessingResult.DENY;
+            }
+
+            return blockState;
+        }
+
+        @Override
+        public List<ItemStack> getRequiredItems(@NotNull final World world, @NotNull final BlockPos pos, @NotNull final IBlockState blockState, @Nullable final NBTTagCompound tileEntityData, final boolean complete)
+        {
```

Comment:
### Warnings:
- Found duplicated code in src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java